apps/console/src/features/organizations/components/edit-organization-role/organization-role-permission.tsx

```diff
@@ -0,0 +1,291 @@
+     * Permissions to hide.
+     */
+    permissionsToHide?: string[];
+    /**
+     * Specifies if the form is submitting
+     */
+    isSubmitting?: boolean;
+}
+
+/**
+ * Component to create the permission tree structure from the give permission list.
+ *
+ * @param props props containing event handlers for permission component
+ */
+export const OrganizationPermissionList: FunctionComponent<OrganizationPermissionListProp> =
+    (props: OrganizationPermissionListProp): ReactElement => {
+
+        const {
+            isReadOnly,
+            emphasize,
+            triggerSubmit,
+            onSubmit,
+            roleObject,
+            isEdit,
+            initialValues,
+            isRole,
+            isSubmitting,
+            permissionsToHide,
+            ["data-testid"]: testId
+        } = props;
+
+        const { t } = useTranslation();
+
+        const [ permissions, setPermissions ] = useState<TreeNode[]>([]);
+        const [ checkedPermission, setCheckedPermission ] = useState<TreeNode[]>([]);
+        const [ previouslyCheckedKeys, setPreviouslyCheckedKeys ] = useState<string[]>([]);
+        const [ defaultExpandedKeys, setDefaultExpandKeys ] = useState<string[]>([]);
+        const [ isPermissionsLoading, setIsPermissionsLoading ] = useState<boolean>(true);
+
+        useEffect(() => {
+            const checkedNodes: TreeNode[] = [];
+
+            OrganizationRoleManagementUtils.getAllOrganizationLevelPermissions(permissionsToHide)
+                .then((permissionTree: TreeNode[]) => {
+                    setPermissions(permissionTree);
+                    setDefaultExpandKeys([ permissionTree[0].key.toString() ]);
+                    setIsPermissionsLoading(false);
+                });
+
+            if (initialValues && initialValues.length > 0) {
+                const previousFormCheckedKeys: string[] = [];
+
+                initialValues.forEach(initialKey => {
+                    previousFormCheckedKeys.push(initialKey.key.toString());
+                });
+                setPreviouslyCheckedKeys(previousFormCheckedKeys);
+                previouslyCheckedKeys?.forEach(key => {
+                    checkedNodes.push(getNodeByKey(key, permissions));
+                });
+                setCheckedPermission(checkedNodes);
+            }
+
+            if (isRole && roleObject) {
+                setPreviouslyCheckedKeys(roleObject.permissions);
+                previouslyCheckedKeys?.forEach(key => {
+                    checkedNodes.push(getNodeByKey(key, permissions));
+                });
+                setCheckedPermission(checkedNodes);
+            }
+        }, [ permissions.length > 0 ]);
+
+        /**
+         * Util function to disable checking of a node and it's children.
+         *
+         * @param permissionNodes - array of permission nodes
+         * @param state - disable state
+         */
+        const disableTreeNode = (permissionNodes: TreeNode[], state: boolean) => {
```

Comment:
shall we add the return type

Reply:
+1

Reply:
Removed this function in as unused in https://github.com/wso2/identity-apps/pull/3617/commits/630fa8dda5d13075f2d424b2493249572e255217